hyundai_kia_connect_api/KiaUvoApiCA.py

```diff
@@ -586,34 +589,38 @@ def stop_climate(self, token: Token, vehicle: Vehicle) -> str:
         response = response.json()
 
         _LOGGER.debug(f"{DOMAIN} - Received stop_climate response: {response}")
-        return response_headers["transactionId"]
+        vehicle.last_action_id = response_headers["transactionId"]
+        return vehicle.last_action_id
 
-    def check_last_action_status(
-        self, token: Token, vehicle: Vehicle, action_id: str
-    ) -> bool:
+    def check_last_action_status(self, token: Token, vehicle: Vehicle) -> bool:
         url = self.API_URL + "rmtsts"
         headers = self.API_HEADERS
         headers["accessToken"] = token.access_token
         headers["vehicleId"] = vehicle.id
-        headers["transactionId"] = action_id
-        headers["pAuth"] = self._get_pin_token(token, vehicle)
+        headers["transactionId"] = vehicle.last_action_id
+        headers["pAuth"] = vehicle.last_pauth
         response = requests.post(url, headers=headers)
         response = response.json()
 
-        last_action_completed = (
-            response["result"]["transaction"]["apiStatusCode"] != "null"
-        )
+        _LOGGER.debug(f"{DOMAIN} - Last action_status API response: {response}")
+
+        api_result = response["result"]["transaction"]["apiResult"]
+        last_action_completed = api_result != "P"
         if last_action_completed:
```

Comment:
```
        last_action_completed = api_result != "P"
        if last_action_completed:
```
Unless I'm reading it wrong, these lines are not necessary.
Testing the value of `api_result` is enough, and you're already doing it in the line that follows.
If we want to test for "P" then I would consider using a simple  `else if` block after the one for "C".

By the way, what does "P" stand for? "Pending"?

Reply:
The naming convention here is a bit confusing. The official API responds with P if the call has not yet been completed,  probably Pending or maybe Progress. It responds with C if the call completed successfully and F if the call failed. Along with C it also provides the latest vehicle data.

Since our API specification calls for check_last_action_status to respond with a boolean value, I have it responding True for both success and failure, and False for pending.

I should have probably structured it like this, a bit easier to follow:

```
        api_result = response["result"]["transaction"]["apiResult"]
        if api_result == "C":
          state = {"status": response["result"]["vehicle"]}
          self._update_vehicle_properties_base(vehicle, state)
        return api_result != "P"
```

Reply:
Maybe we adjust away from boolean?

Reply:
I think that would make sense sense- it sounds like we have four potential valid states: pending, success, fail and timeout.

In CA we get a friendly message along with a failure as well, for example apiFailureReason: "We cannot process your request. Please verify that your vehicle's doors, hood and trunk are closed and locked", it may make sense to pass that along as well.

I don't know if CA has a timeout state, but I suppose in practice we would handle failure and timeout in a similar manner. Do we have anyone who can verify how things look in US?

Reply:
> it may make sense to pass that along as well.

In EU the message in the response is generic and quite useless: "cannot run command. please check vehicle status" ([source](https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/discussions/272#discussion-4792651)). I'm not sure the CA message is much better?
If it's generic I dont think it's useful to return it.